homeassistant/components/zone/__init__.py

```diff
@@ -362,10 +366,17 @@ async def async_added_to_hass(self) -> None:
         await super().async_added_to_hass()
         person_domain = "person"  # avoid circular import
         persons = self.hass.states.async_entity_ids(person_domain)
-        object_id = split_entity_id(self.entity_id)[1]
         for person in persons:
             state = self.hass.states.get(person)
-            if state and state.state == object_id:
+            if (
+                state is None
+                or (latitude := state.attributes.get(ATTR_LATITUDE)) is None
+                or (longitude := state.attributes.get(ATTR_LONGITUDE)) is None
+                or (accuracy := state.attributes.get(ATTR_GPS_ACCURACY)) is None
```

Comment:
I might be totally wrong and in that case apologies - but I'm wondering how this works with cases where device_trackers are not providing lat, long and accuracy (e.g., as enabled by https://github.com/home-assistant/core/pull/62243 for mobile_app trackers for example)

Reply:
Actually, I just tried this out, creating a person with a single device tracker (Wi-Fi based). This person has state == home, but does not have latitude or longitude in attributes.

Reply:
A person or device can have an arbitrary state, but that doesn't tie them into zones.
Zones are purely based on coordinates.

For example, triggers and conditions on zones don't work without coordinates either.

Reply:
Yes, fully agree that zones by definition have coordinates, and this makes sense!

However, conditions on zones do work without coordinates on the person/ device_tracker, they are just comparing states. In my opinion, we should do the same with the zone counter (i.e., state)

**My example:**

- One zone called home, with coordinates.
- 1 person, with a single device_tracker (e.g., wifi-based one, iOS App 2022.02 with name-only tracker, etc.) which is NOT reporting coordinates 

![Screenshot 2022-01-27 at 14 32 47](https://user-images.githubusercontent.com/26942220/151369227-d8fa06b8-8bef-4f8d-8759-508dc6f7b32a.png)

- state of this person is the name of the zone, e.g., `home`, `not_home`
- state attributes of the person do not contain coordinates (because the device tracker is not reporting coordinates)


**(Existing) Zone condition**
```
condition:
  alias: "Person at home"
  condition: zone
  entity_id: person.name
  zone: zone.home
```
The condition is true, although the person is not reporting coordinates

**(New) Zone state**
As per my understanding of this PR (but please correct me if I'm wrong), the person would not be included in the count that's the state of the zone, although it is being considered home by the zone condition. I think this would be inconsistent.

Reply:
Apologies - I believe the automation I tested this with was faulty. You are fully right.

Reply:
According to the condition logic, it raises a condition error when a zone condition is used on an entity without lat/long:

https://github.com/home-assistant/core/blob/079439812037c8904fedef4554ca9554936521dc/homeassistant/helpers/condition.py#L816-L831

Reply:
Yes, you are fully right @frenck - I've hidden my post above. Sorry for the confusion!